numbat/src/help.rs

```diff
@@ -0,0 +1,146 @@
+
+            match interpreter_result {
+                InterpreterResult::Value(value) => {
+                    let type_ = statements.last().map_or(m::empty(), |s| {
+                        if let crate::Statement::Expression(e) = s {
+                            let type_ = e.get_type();
+
+                            if type_ == Type::scalar() {
+                                m::empty()
+                            } else {
+                                m::dimmed("    [")
+                                    + e.get_type().to_readable_type(&registry)
+                                    + m::dimmed("]")
+                            }
+                        } else {
+                            m::empty()
+                        }
+                    });
+
+                    full_output += statement_output.lock().unwrap().iter().fold(
+                        m::empty(),
+                        |accumulated_mk, single_line| {
+                            accumulated_mk
+                                + m::nl()
+                                + m::whitespace("  ")
+                                + single_line.clone()
+                                + m::nl()
+                        },
+                    ) + m::nl()
+                        + m::whitespace("    ")
+                        + m::operator("=")
+                        + m::space()
+                        + value.pretty_print()
+                        + type_
+                        + m::nl();
+                }
+                InterpreterResult::Continue => {
+                    full_output += statement_output.lock().unwrap().iter().fold(
+                        m::empty(),
+                        |accumulated_mk, single_line| {
+                            accumulated_mk
+                                + m::nl()
+                                + m::whitespace("  ")
+                                + single_line.clone()
+                                + m::nl()
+                        },
+                    );
+                }
+                InterpreterResult::Exit(_exit_status) => {
+                    println!("Interpretation Error.");
+                }
+            }
+        }
+        Err(NumbatError::ResolverError(e)) => {
+            context.print_diagnostic(e.clone());
+        }
+        Err(NumbatError::NameResolutionError(
+            e @ (NameResolutionError::IdentifierClash { .. }
+            | NameResolutionError::ReservedIdentifier(_)),
+        )) => {
+            context.print_diagnostic(e);
+        }
+        Err(NumbatError::TypeCheckError(e)) => {
+            context.print_diagnostic(e);
+        }
+        Err(NumbatError::RuntimeError(e)) => {
+            context.print_diagnostic(e);
+        }
+    }
+
+    full_output
+}
+
+pub fn help_markup(pretty_print: bool) -> m::Markup {
+    let mut output = m::nl()
+        + m::keyword("numbat")
+        + m::space()
+        + m::text(env!("CARGO_PKG_DESCRIPTION"))
```

Comment:
I like that you want to avoid the duplication here, but I'd rather see a proper sentence starting with "Numbat is a statically …", instead of "numbat A statically …". Let's just copy that sentence here?

A newline between the description and the first example would also be great.